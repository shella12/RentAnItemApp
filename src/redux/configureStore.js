import { configureStore } from '@reduxjs/toolkit';
import favoriteReducer from './favorites/favoriteReducer';
import housesSlice from './house/house';

const store = configureStore({
  reducer: {
    housesSlice,
    favorite: favoriteReducer,
  },
});

export default store;
