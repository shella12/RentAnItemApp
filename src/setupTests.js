import '@testing-library/jest-dom';
import fetchMock from 'fetch-mock';

export const testListHouses = [{
  id: 1, name: 'Pent house', price: '300000.0', picture: 'https://github.com/shella12/RentAnItemBackend/assets/44798044/5bb4ea2c-4ae5-45b8-9083-aeafbfbcf287', description: "This house has 5 rooms upstairs and 4 rooms downstairs. Home design emphasizes clean lines and geometric shapes.\n         The magic thing about this houe is that it feels good to leave, and it feels even better to come back.\n         Modern Conveniences and Green Features. Well-Placed Bathrooms. Quality Construction with Distinct Architecture. \n         Plenty of Storage Space. It's important to have designated storage areas and large closets.", owner_name: 'susan perry', created_at: '2023-05-12T12:38:59.989Z', updated_at: '2023-05-12T12:38:59.989Z',
}, {
  id: 6, name: 'Mansion', price: '870000.0', picture: 'https://github.com/shella12/RentAnItemBackend/assets/44798044/ac143109-1ae3-4885-a650-808b3bdb6cda', description: "This house has 5 rooms upstairs and 4 rooms downstairs. Home design emphasizes clean lines and geometric shapes.\n         The magic thing about this houe is that it feels good to leave, and it feels even better to come back.\n         Modern Conveniences and Green Features. Well-Placed Bathrooms. Quality Construction with Distinct Architecture. \n         Plenty of Storage Space. It's important to have designated storage areas and large closets.", owner_name: 'Kim buffey', created_at: '2023-05-12T12:39:00.010Z', updated_at: '2023-05-12T12:39:00.010Z',
}];

export const testListFavoriteHouses = [
  {
    id: 1,
    name: 'Pent house',
    price: '300000.0',
    picture: 'https://github.com/shella12/RentAnItemBackend/assets/44798044/5bb4ea2c-4ae5-45b8-9083-aeafbfbcf287',
    description: "This house has 5 rooms upstairs and 4 rooms downstairs. Home design emphasizes clean lines and geometric shapes.\n         The magic thing about this houe is that it feels good to leave, and it feels even better to come back.\n         Modern Conveniences and Green Features. Well-Placed Bathrooms. Quality Construction with Distinct Architecture. \n         Plenty of Storage Space. It's important to have designated storage areas and large closets.",
    owner_name: 'susan perry',
    created_at: '2023-05-12T12:38:59.989Z',
    updated_at: '2023-05-12T12:38:59.989Z',
  },
];

export const fetchFavoriteHouseData = (userID) => {
  fetchMock.getOnce(`https://renthousebackend.onrender.com/api/v1/users/${userID}/favorite_houses`, {
    headers: { 'content-type': 'application/json' },
    body: testListFavoriteHouses,
  });
};

export const fetchHouseData = () => {
  fetchMock.getOnce('https://renthousebackend.onrender.com/api/v1/houses', {
    headers: { 'content-type': 'application/json' },
    body: testListHouses,
  });
};
