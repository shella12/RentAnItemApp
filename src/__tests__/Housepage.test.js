import renderer, { act } from 'react-test-renderer';
import '@testing-library/jest-dom';
import React from 'react';
import { Provider } from 'react-redux';
import { waitFor } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import { fetchHouseData } from '../setupTests';
import configureStore from '../redux/configureStore';
import Home from '../pages/Homepage';

const store = configureStore;
fetchHouseData();

jest.mock('../pages/Homepage.js', () => {
  const MockedHomepage = () => (
    <>
      <h1>Houses</h1>
      <p className="flex-center empty-list">No Houses: List Empty</p>
    </>
  );
  MockedHomepage.displayName = 'Homepage';
  return MockedHomepage;
});

describe('test render', () => {
  test('Add House page should match snapshoot', async () => {
    let tree;
    act(() => {
      tree = renderer.create(
        <React.StrictMode>
          <Provider store={store}>
            <MemoryRouter initialEntries={['/houses']}>
              <Home />
            </MemoryRouter>
          </Provider>
        </React.StrictMode>,
      );
    });

    await waitFor(() => {
      expect(tree.toJSON()).toMatchSnapshot();
    });
  });
});
