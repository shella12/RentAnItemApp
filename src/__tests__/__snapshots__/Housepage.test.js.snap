// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test render Add House page should match snapshoot 1`] = `
Array [
  <h1>
    Houses
  </h1>,
  <p
    className="flex-center empty-list"
  >
    No Houses: List Empty
  </p>,
]
`;
